### PR TITLE
WIP: Add ability to run the reports from the IDE without having to startup a server 

### PR DESCRIPTION
**Pull request by ssmusoke**

### CODE REVIEW

**Comment by ssmusoke** [comment]
@dkayiwa Good afternoon I am running into this error a Null Pointer Exception when running the reports https://gist.github.com/ssmusoke/01a5f0ebd3f0a2327c2eb7dd0cbc6c07

**Comment by dkayiwa** [comment]
@ssmusoke is this as simple to reproduce as mvn clean install?

**Comment by ssmusoke** [comment]
@dkayiwa Yes it is that simple :-)

**Comment by dkayiwa** [comment]
@ssmusoke try making StandaloneContextSensitiveTest's getRuntimeProperties() like this: https://pastebin.com/Ri0JMRnx

**Comment by ssmusoke** [comment]
@dkayiwa Please can you help me work your magic on this error too https://gist.github.com/ssmusoke/19bb0e72d1be54e83d2566c62b6fd669

**Comment by dkayiwa** [comment]
Are your changes committed?

**Comment by ssmusoke** [comment]
@dkayiwa Apologies had committed not pushed - now done

**Comment by dkayiwa** [comment]
This error is different from the one in your pastebin link.

**Comment by ssmusoke** [comment]
@dkayiwa What are you getting?

**Comment by dkayiwa** [comment]
Look at travis.

**Comment by ssmusoke** [comment]
@dkayiwa That is the error that I am having where I need your magic to make it go away

**Comment by dkayiwa** [comment]
Which is different from the one in the paste link. Correct?

**Comment by ssmusoke** [comment]
@dkayiwa its the same error

**Comment by dkayiwa** [comment]
@ssmusoke first step, can you remove the web related dependencies from your api pom.xml file?

**Comment by ssmusoke** [comment]
@dkayiwa such as which ones? When I remove openmrs-web the StandardContextSensitiveTest does not run

File: api/src/test/java/org/openmrs/module/ugandaemrreports/StandaloneContextSensitiveTest.java [hunk 1]
```diff
@@ -22,15 +22,13 @@ public Boolean useInMemoryDatabase() {
 	 */
 	@Override
 	public Properties getRuntimeProperties() {
-		Properties p = super.getRuntimeProperties();
+		System.setProperty("databaseUrl", "jdbc:mysql://localhost:3306/openmrs?autoReconnect=true&sessionVariables=storage_engine%3DInnoDB&useUnicode=true&characterEncoding=UTF-8");
+		System.setProperty("databaseUsername", "openmrs");
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/module/ugandaemrreports/StandaloneContextSensitiveTest.java]: @slubwama this method as you see it and also override the useInMemoryDatabase() method to return false.

**Comment by ssmusoke** [comment]
@dkayiwa I have removed all web dependencies from the api submodule, new error is https://gist.github.com/ssmusoke/ed3af9cf284deed22342a0abedf5ea6a

**Comment by dkayiwa** [comment]
Add the reporting-api-1.9 dependency

**Comment by dkayiwa** [comment]
@ssmusoke what error are you getting?

**Comment by ssmusoke** [comment]
@dkayiwa This one https://gist.github.com/ssmusoke/3b97c59bd142a8e9f5480dea48d90645

Which seems to be caused by sample data from this line https://github.com/openmrs/openmrs-contrib-liquibaserunner/blob/e6337520dc679a70864d735fc599ff8e736a40fd/test/src/test/resources/openmrs-1.6.x/1.6.6.27476-2012.05.28-clean.sql#L935

This is despite adding @SkipBaseSetup to the TestSetup106A1AReport class

**Comment by dkayiwa** [comment]
Your paste link is different from the error in travis.

**Comment by ssmusoke** [comment]
@dkayiwa This is because there is no connection - however if you run it locally with a database, you will get the error I am getting - which is the key for this test

**Comment by dkayiwa** [comment]
Oh i see. Let me try run locally.

**Comment by ssmusoke** [comment]
@dkayiwa Good morning any success?

**Comment by dkayiwa** [comment]
@ssmusoke good morning. Did you see my question above?

**Comment by ssmusoke** [comment]
@dkayiwa you should expect to get this error https://gist.github.com/ssmusoke/3b97c59bd142a8e9f5480dea48d90645

**Comment by dkayiwa** [comment]
Did you at any one time get this? Caused by: java.sql.SQLException: Connections could not be acquired from the underlying database!

**Comment by ssmusoke** [comment]
@dkayiwa You may need to change your database access credentials here https://github.com/METS-Programme/openmrs-module-ugandaemr-reports/blob/1228ab773b59646e605421d6900cc99ca1193415/api/src/test/java/org/openmrs/module/ugandaemrreports/StandaloneContextSensitiveTest.java#L23

**Comment by dkayiwa** [comment]
Do you get the same error if you mess up your database credentials?

**Comment by ssmusoke** [comment]
@dkayiwa YEs - just like this https://gist.github.com/ssmusoke/f701b2978b8384cf5d4d332720b819ed

**Comment by dkayiwa** [comment]
After running this test, can you check your database and tell me if you still have any data in it?

**Pull request closed without merging**